SalesforceCommerceCloud/pwa-kit
# Add valid token check before using refresh token on login

GUS: https://gus.lightning.force.com/lightning/_classic/%2Fa07EE00000voPZ9YAM

# Description

This PR adds a check on login to see whether an existing access token is still valid. If so, the login will exit early so we do not make an unnecessary login call to SLAS using the refresh token.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Update login function so that there is a check to see whether the existing token is valid
- Reintroduces this.customerId to Auth (after it had been removed from #528). This is needed so we can return a valid customer object from the login function without needing to call SLAS for a customerId. The customer id is stored in storage.

# How to Test-Drive This PR

Reproducing the issue on the main `v2` branch:
- Start the app and open devtools. View local storage (or cookie storage)
- Hard refresh the app. Observe that the refresh token (either `cc-nx` or `cc-nx-g`) is consumed and replaced with a new one

In this branch:
- Follow the steps above. Observe that the refresh token (either `cc-nx` or `cc-nx-g`) stays the same after a hard refresh
- Wait 6 hours (this is the hard timeout for sessions). Do a hard refresh of the app. Observe that the refresh token is used and a new one is stored.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
